CapstoneProject/Assets/Scripts/Core/Puzzle/SpikeBall.cs

```diff
@@ -0,0 +1,59 @@
+using UnityEngine;
+
+public class SpikeBall : MonoBehaviour
+{
+    [SerializeField]
+    private float rotationSpeed = 100f;
+    [SerializeField]
+    private float radius = 2f;
+    [SerializeField]
+    private int numPathPoints = 20;
+    [SerializeField]
+    private GameObject pathPointPrefab;
+
+    private Vector2 _centerPosition;
+    private float angle;
+
+    private void Start()
+    {
+        // Get the center position around which the spike ball will rotate
+        _centerPosition = transform.position;
+        // Initiate the angle
+        angle = 0;
+        // Generate path points
+        GeneratePathPoints();
+    }
+
+    private void Update()
+    {
+        // Update the angle based on the rotation speed
+        angle += rotationSpeed * Time.deltaTime;
+
+        // Calculate the new position based on the angle and radius
+        float x = _centerPosition.x + Mathf.Cos(angle * Mathf.Deg2Rad) * radius;
+        float y = _centerPosition.y + Mathf.Sin(angle * Mathf.Deg2Rad) * radius;
+        Vector2 newPosition = new Vector2(x, y);
```

Comment:
its better in an update loop like this to just have a cached single vector you update the position of to apply each time rather than a constant new Vec2 each frame